File: backend/pricingService/src/main/java/com/nsbm/bunmart/pricing/model/Discount.java
package com.nsbm.bunmart.pricing.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "discounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Discount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "product_id", nullable = false, length = 36)
    private String productId;

    @Column(name = "min_quantity", nullable = false)
    private Integer minQuantity;

    @Column(nullable = false, length = 20)
    private String type; // PERCENT, FIXED

    @Column(nullable = false, precision = 19, scale = 4)
    private BigDecimal value;

    @Column(length = 500)
    private String description;

    @Column(name = "is_active")
    private Boolean isActive;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (isActive == null) isActive = true;
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/pricing/models/Discount.ts
export interface DiscountRule {
  id?: number;
  productId?: string;
  campaignId?: number;
  type: string; // 'PERCENTAGE' or 'FIXED'
  value: number; // Percentage or fixed amount in LKR
  description?: string;
  isActive?: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateDiscountDto {
  productId?: string;
  campaignId?: number;
  type: string;
  value: number; // Percentage or fixed amount in LKR
  description?: string;
  isActive?: boolean;
}